src/language-js/comments.js

```diff
@@ -884,6 +892,30 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function handleLogicalExpression(
+  text,
+  enclosingNode,
+  precedingNode,
+  followingNode,
+  comment,
+  options
+) {
+  if (enclosingNode && enclosingNode.type === "LogicalExpression") {
```

Comment:
This bug happens for `BinaryExpression`, not only `LogicalExpression`.

[Playground Link](https://deploy-preview-7787--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEA3AhgJwATuwXmwCNsBqbAegCorsABGATwAc5tgBnGTASygHMAvtioVsAHSjZp2MAG4QAGhARmMHtA7JQWTBADuABSwItKdABt96RluVFM6MAGs4MAMrMnffsm4BXOGUACxgAWwsAdWCeeA4vMDh3U1ieVFjGZHAOOxA+DjhMGENHfjD0ZAAzSwLlACsOAA8AIUcXN3d0MLgAGT44KpqgkAbG9x8LOABFfwh4QYtakC9MAswsonQiOAslZd5YSJ4AExhg5AAOAAZlZj0CyMdmLLu4NdQB5QBHWfgS1TMIHQHAAtFA4HBjpC9pg4D8eLCSugyhUkNVFsMCmEeH5MIFlBwJtNfgM0UNlDAtkdTuckAAmCmOHgWHwAYQgYXKWTeAFY9v4CgAVLZmdFLVCBACSUChsHcYF4agAgjL3ExJgsCoJBEA)

**Input:**

```js
var a = b + /** @type {string} */ 
    c;
```

**Output:**

```js
var a = b /** @type {string} */ + c;
```

Reply:
Thanks. I will add `handleBinaryExpression` function.